src/logic/currencyValues/store/middleware/index.ts

```diff
@@ -1,41 +1,18 @@
 import fetchCurrencyRate from 'src/logic/currencyValues/store/actions/fetchCurrencyRate'
-import { SET_CURRENCY_BALANCES } from 'src/logic/currencyValues/store/actions/setCurrencyBalances'
-import { SET_CURRENCY_RATE } from 'src/logic/currencyValues/store/actions/setCurrencyRate'
 import { SET_CURRENT_CURRENCY } from 'src/logic/currencyValues/store/actions/setSelectedCurrency'
-import { currencyValuesSelector } from 'src/logic/currencyValues/store/selectors'
-import { saveCurrencyValues } from 'src/logic/currencyValues/store/utils/currencyValuesStorage'
-import { AVAILABLE_CURRENCIES, CurrencyRateValue } from '../model/currencyValues'
-import { Map } from 'immutable'
 
-const watchedActions = [SET_CURRENT_CURRENCY, SET_CURRENCY_RATE, SET_CURRENCY_BALANCES]
+const watchedActions = [SET_CURRENT_CURRENCY]
 
 const currencyValuesStorageMiddleware = (store) => (next) => async (action) => {
   const handledAction = next(action)
   if (watchedActions.includes(action.type)) {
-    const state = store.getState()
     const { dispatch } = store
     switch (action.type) {
       case SET_CURRENT_CURRENCY: {
         const { safeAddress, selectedCurrency } = action.payload
         dispatch(fetchCurrencyRate(safeAddress, selectedCurrency))
         break
       }
-      case SET_CURRENCY_RATE:
```

Comment:
Hi @mikheevm, want to ask you why did you removed this. Because while I was working on [this ticket](https://github.com/gnosis/safe-react/issues/1312) I found that this was part of the problem. Also, now the storage of the selected currency is broken, for example, if you choose EUR as your preferred currency and reload the page, you will see USD again

Reply:
Hey, as I communicated before we don't need to store the balances locally as the rates frequently change and it's fine to fetch this again on app startup. I didn't expect that `SET_CURRENCY_RATE` action is capable of storing currently selected currency, I thought `SET_CURRENCY_CURRENCY` action was responsible for that because that's literally the action name. Sorry for this.

Reply:
No problem! thanks for the clarification :)